apps/jobs/views.py

```diff
@@ -2073,3 +2077,154 @@ def get_github_badge_data(
         else:
             data["message"] = f"{challenge_obj.title}"
     return Response(data, status=http_status_code)
+
+
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_presigned_url_for_submission(
+    request, challenge_pk, challenge_phase_pk
+):
+    # Abstract out the validation
+    try:
+        challenge = Challenge.objects.get(pk=challenge_pk)
+    except Challenge.DoesNotExist:
+        response_data = {"error": "Challenge does not exist"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if not is_user_a_host_of_challenge(request.user, challenge_pk) and not challenge.approved_by_admin:
```

Comment:
@KhalidRmb do the challenge approval check in the code block at line 2125. The check is redundant `if not is_user_a_host_of_challenge(request.user, challenge_pk)`. See the submission view for how to order these checks